api/config.yaml

```diff
@@ -233,6 +233,62 @@ yarnaudit:
   default: true
   timeOutInSeconds: 360
 
+spotbugs:
+  name: spotbugs
+  image: huskyci/spotbugs
+  imageTag: "4.0.0-beta4"
+  cmd: |+
+    mkdir -p ~/.ssh &&
+    echo 'GIT_PRIVATE_SSH_KEY' > ~/.ssh/huskyci_id_rsa &&
+    chmod 600 ~/.ssh/huskyci_id_rsa &&
+    echo "IdentityFile ~/.ssh/huskyci_id_rsa" >> /etc/ssh/ssh_config &&
+    echo "StrictHostKeyChecking no" >> /etc/ssh/ssh_config &&
+    git clone -b %GIT_BRANCH% --single-branch %GIT_REPO% code --quiet 2> /tmp/errorGitCloneSpotBugs
+    if [ $? -eq 0 ]; then
+       cd code
+       if [ -f "pom.xml" ]; then
+           mv ../code /tmp/code
+           cd /tmp/code
+           project_type=$(cat pom.xml|grep packaging|cut -d'<' -f2|cut -d'>' -f2)
+           bash /usr/local/bin/mvn-entrypoint.sh 2> /tmp/errorMavenBuild 1> /dev/null
+           if [ $? -eq 0 ]; then
+               if [[ "$project_type" = "war" ]]; then
+                  # WAR
+                   war_file=$(ls -lS target/*.war|head -n1|cut -d'/' -f2|sed -e "s/.war//")
+                   mv target/$war_file /tmp/needToBeScanned
+               else
+                  # JAR
+                   mkdir /tmp/needToBeScanned
+                   cp target/*.jar /tmp/needToBeScanned/
+               fi
+               java -jar /opt/spotbugs/lib/spotbugs.jar -textui -quiet -xml -exitcode -bugCategories SECURITY -exclude /opt/spotbugs/exclude.xml -pluginList /opt/findsecbugs-plugin-1.9.0.jar /tmp/needToBeScanned
+           else
+               echo "ERROR_RUNNING_MAVEN_BUILD"
+               cat /tmp/errorMavenBuild
+           fi
+       elif [ -f "build.gradle" ]; then
+           mv ../code /tmp/code
+           cd /tmp/code
+           /opt/gradle/bin/gradle -p /tmp/code build 2> /tmp/errorGradleBuild 1> /dev/null
+            if [ $? -eq 0 ]; then
+               mv build /tmp/needToBeScanned
+               java -jar /opt/spotbugs/lib/spotbugs.jar -textui -quiet -xml -exitcode -bugCategories SECURITY -exclude /opt/spotbugs/exclude.xml -pluginList /opt/findsecbugs-plugin-1.9.0.jar /tmp/needToBeScanned
```

Comment:
We need to remove `-exitcode` from this command as it is returning an exit code equals to 1, which is bad for huskyCI. 🧐